### PR TITLE
TMF 103 (frontend) pagination of posts in a category and the dynamic routing

### PR DESCRIPTION
**Pull request by zzyang1994**

### CODE REVIEW

**Comment by Qiming-Liu** [comment]
https://thinkmoreforum-git-tmf-103-frontend-paginatio-a43e15-qiming-liu.vercel.app/category/Default%20Category

为啥是空的

File: components/Post/PostCard.js [hunk 1]
```diff
@@ -82,6 +91,15 @@ const PostCard = (props) => {
               By {authorName} • {createTimeStamp}
             </Typography>
           </Box>
+          <Typography
+            align="right"
+            color="textSecondary"
+            sx={{ flexGrow: 1 }}
+            variant="body2"
+          >
+            {`${viewCount} views`} • {`${commentCount} comments`} •{' '}
```
Comment (devharris7) [inline components/Post/PostCard.js]: why not put everything in a string?

File: pages/category/[categoryTitle]/index.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  const sizePerPage = 10;
+  const [totalPages, setTotalPages] = useState(0);
+  useEffect(() => {
+    const fetchData = async () => {
+      const responsePosts = await getPostsByCategoryTitle(
+        categoryTitle,
+        currentPage,
+        sizePerPage,
+      );
+      const responseTotalPages = await getPagesByCategoryTitle(categoryTitle);
```
Comment (devharris7) [inline pages/category/[categoryTitle]/index.js]: const {data} =

File: pages/category/[categoryTitle]/index.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        const timeStamp = new Date(post.createTimestamp);
+        const createDate = timeStamp.toLocaleDateString('en-AU');
+        const createTime = timeStamp.toLocaleTimeString('en-AU');
+        const concatedDateTime = `${createDate.toString()} ${createTime.toString()}`;
+        return (
+          <PostCard
+            key={post.id}
+            generatedUrl={`/category/${categoryTitle}/post/${post.id}`}
+            authorAvatar={
+              post.postUsers.profileImg
```
Comment (devharris7) [inline pages/category/[categoryTitle]/index.js]: consider if postUsers or profileImg will be undefined

File: pages/category/[categoryTitle]/index.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+      <NextLink href="/" passHref>
+        <Button component="a" startIcon={<ArrowLeftIcon fontSize="small" />}>
+          Back to Home
+        </Button>
+      </NextLink>
+      <Typography variant="h3" sx={{ mt: 3 }}>
+        {categoryTitle}
+      </Typography>
+      <Divider sx={{ my: 3 }} />
+      {posts.map((post) => {
```
Comment (devharris7) [inline pages/category/[categoryTitle]/index.js]: ```map(({id, postUsers,headImage,title}) ```like this

**Pull request merged**